core/modules/search/search.module

```diff
@@ -111,13 +111,103 @@ function search_block_info() {
   return $blocks;
 }
 
+/**
+ * Implements hook_block_configure().
+ */
+function search_block_configure($delta = '', $settings = array()) {
+  $form = array();
+  if ($delta == 'form') {
+    $search_settings = !empty($settings['search'])? $settings['search'] : array();
+    $search_settings += array(
+      'label' => '',
+      'label_hide' => TRUE,
+      'placeholder' => '',
+      'button_text' => 'Search',
+    );
+
+    $form['search'] = array(
+      '#type' => 'fieldset',
+      '#title' => t('Search form settings'),
+      '#collapsible' => TRUE,
+    );
+
+    $form['search']['label'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Search box label'),
+      '#default_value' => check_plain($search_settings['label']),
+    );
+    $form['search']['label_hide'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Hide the search box label'),
+      '#default_value' => $search_settings['label_hide'],
+    );
+    $form['search']['placeholder'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Placeholder text'),
+      '#default_value' => check_plain($search_settings['placeholder']),
+      '#description' => t('This text will appear in the search box until text is entered.'),
+    );
+    $form['search']['button_text'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Button text'),
+      '#default_value' => check_plain($search_settings['button_text']),
+      '#description' => t('Leave this field empty to hide the submit button.'),
+    );
+  }
+  return $form;
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function search_form_layout_block_configure_form_alter(&$form, &$form_state) {
```

Comment:
Another way would have been for `hook_block_view()` set `$block['subject'] = '';`
Users can set a block title in the block config UI title settings dropdown if they want. That would need an update hook to explicitly reset existing blocks with `"title_display": "default"` to `"title_display": "custom"` and `"title": "Search"` (translated).

Reply:
Hm, I like your way better. It's much cleaner. The update hook might be tricky though. I think maybe I'll move that to another issue so we can wrap this up and get it done :) 
Tah Dah: https://github.com/backdrop/backdrop-issues/issues/3476

Reply:
I've removed the form alter from this PR. Will work on that over in https://github.com/backdrop/backdrop-issues/issues/3476